includes/class-img-converter.php

```
@@ -344,30 +344,44 @@ private function is_animated_webp( $file ) {
 	 * @since 1.0.0
 	 */
 	public static function get_img_path( string $source_image, string $format = 'webp' ): string {
-		$info = pathinfo( $source_image );
+		if ( strpos( $source_image, home_url() ) !== 0 && strpos( $source_image, '/' ) !== 0 ) {
+			return $source_image;
+		}
 
-		$new_file_name = $info['filename'] . '.' . $format;
+		// Use Util::get_local_path to get a clean local path from URL or existing path.
+		$local_path = Util::get_local_path( $source_image );
```

PHPMD (2.15.0)

[error] Avoid using static access to class 'PerformanceOptimise\Inc\Util' in method 'get_img_path'. (undefined)

(StaticAccess)